tasks/git/src/main/java/com/walmartlabs/concord/plugins/git/client/GitHubClient.java

```diff
@@ -63,17 +68,38 @@ public GitHubClient(UUID txId, GitHubApiInfo apiInfo) {
                 .build();
     }
 
-    @SuppressWarnings("unchecked")
     public Map<String, Object> singleObjectResult(String method, String path, Object body) throws IOException, InterruptedException, URISyntaxException {
-        var response = sendRequest(method, path, body);
-        var parsed = parseResponseAsObject(response);
-        if (parsed instanceof Map) {
-            return (Map<String, Object>) parsed;
+        var response = sendRequest(method, path, Map.of(), body);
+        return parseResponseAs(response,  OBJECT_TYPE);
+    }
+
+    public void forEachPage(String path, Map<String, String> params, int pageSize, PageHandler handler)
+            throws IOException, InterruptedException, URISyntaxException {
+
+        Objects.requireNonNull(path, "'path' must not be null");
+        Objects.requireNonNull(params, "'params' must not be null");
+
+        var effectiveParams = new HashMap<>(params);
+        effectiveParams.put("per_page", String.valueOf(pageSize));
+
+        var apiUrl = buildApiUrl(apiInfo.baseUrl(), path);
+        var requestUrl = appendParamsToUrl(apiUrl, effectiveParams);
+        while (requestUrl != null) {
```

Comment:
When `requestUrl` is null? Should it instead be something like

```java
while (true) {
  var keepGoing = handler.onPage(...)
  if (!keepGoing) {
    break;
  }
  ...etc
}
```

Reply:
```
requestUrl = parseNextLink(response.headers().firstValue("Link").orElse(null));
```